js/geo/lng_lat.js

```diff
@@ -82,9 +82,11 @@ LngLat.prototype.toString = function () {
 LngLat.convert = function (input) {
     if (input instanceof LngLat) {
         return input;
-    }
-    if (Array.isArray(input)) {
+    } else if (input && input.lng && input.lat) {
+        return input;
```

Comment:
If given `{lng, lat}` object, it should return a `LngLat` instance rather than the object.